src/index.js

```diff
@@ -46,14 +66,81 @@ window.addEventListener('load', () => {
 
 const input = document.getElementById('new-item');
 input.addEventListener('keypress', (e) => {
-  if (e.key === 'Enter') {
-    if (input.value !== '') {
-      tasksArr.push({
-        description: input.value,
-        completed: false,
-        index: tasksArr[tasksArr.length - 1].index + 1,
-      });
-      input.value = '';
+  if (e.key === 'Enter' && input.value) {
+    addToArr(input.value, tasksArr);
+    input.value = '';
+    appendList(tasksArr.length);
+  }
+});
+
+const enterIcon = document.querySelector('.fa-arrow-turn-down');
+enterIcon.addEventListener('click', () => {
+  if (input.value) {
+    addToArr(input.value, tasksArr);
+    input.value = '';
+    appendList(tasksArr.length);
+  }
+});
+
+const refreshIcon = document.querySelector('.fa-arrows-rotate');
+if (refreshIcon) {
+  refreshIcon.addEventListener('click', () => {
+    generateList();
+  });
+}
+
+const saveFunc = (e) => {
+  const editTextBox = e.target.previousElementSibling.previousElementSibling
+    .previousElementSibling;
+  if (editTextBox.value) {
+    e.target.style.display = 'none';
+
+    e.target.previousElementSibling.style.display = 'block';
+    const delElementDOM = document.getElementsByClassName('todo-li-elements');
+    const delElementArr = e.target.parentNode;
+    let elementID;
+    for (let j = 0; j < delElementDOM.length; j += 1) {
+      if (delElementDOM[j] === delElementArr) {
+        elementID = j + 1;
+      }
+    }
+    updateArr(elementID, editTextBox.value, tasksArr);
+    editTextBox.readOnly = true;
+  }
+};
+
+document.addEventListener('click', (e) => {
+  if (e.target.className === 'fa-solid fa-trash-can fa-xs') {
+    const delElementDOM = document.getElementsByClassName('todo-li-elements');
+    const delElementArr = e.target.parentNode;
+    let elementID;
+    for (let j = 0; j < delElementDOM.length; j += 1) {
+      if (delElementDOM[j] === delElementArr) {
+        elementID = j + 1;
+      }
     }
+    e.target.parentNode.remove();
+    removeFromArr(elementID, tasksArr);
   }
+
+  // Edit button ==================
+  if (e.target.className === 'fa-regular fa-pen-to-square fa-xs') {
+    e.target.nextElementSibling.style.display = 'block';
+    e.target.style.display = 'none';
+    const editTextBox = e.target.previousElementSibling.previousElementSibling;
+    editTextBox.readOnly = false;
+    editTextBox.focus();
+  }
+
+  // Save button ==================
+  if (e.target.className === 'fa-regular fa-floppy-disk fa-xs') {
+    saveFunc(e);
+  }
+});
+
+const clearList = document.querySelector('.clear');
```

Comment:
- [ ] When I click `clear all completed` all the tasks got removed from the array, kindly make sure to fix this so only the completed tasks get deleted

Reply:
Fixed.